### PR TITLE
feat: Add DataTypeMode config

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira:
- https://keboola.atlassian.net/browse/PSGO-531
- https://keboola.atlassian.net/browse/PSGO-532
- https://keboola.atlassian.net/browse/PSGO-533

### CODE REVIEW

**Comment by jachym-tousek-keboola** [comment]
I added NewManifestMode type as enum, similarly to how Mode works. However as is this fails on `fatal error: stack overflow` - see pipeline.

If I change the `%v` [here](https://github.com/keboola/processor-split-table/blob/a7a72518b86a10120599a3c13dcb2d8d2e629da3/internal/pkg/slicer/config/config.go#L187) to `%w` then the stack overflow disappears and it fails with this errror and also breaks unit tests:
```
Error: unexpected value "%!w(config.NewManifestMode=0)" for "new-manifest-mode"
```
Note that adding `--new-manifest-mode columns` doesn't fix this. It seems to crash before even processing the flags / envs.

The only way to fix it I managed to find is to define a default value [here](https://github.com/keboola/processor-split-table/blob/a7a72518b86a10120599a3c13dcb2d8d2e629da3/internal/pkg/slicer/config/config.go#L77-L99) - this is why the same problem doesn't appear for the Mode config - it does have default value. But then of course the flag is not required.

And considering I can't use breakpoints in E2E tests it's very difficult to debug this.

File: README.md [hunk 1]
```diff
@@ -95,14 +95,17 @@ The threshold can be configured by the following flags:
   - Path where the output manifest will be written.
   - The parent directory must exist.
   - The output manifest is a copy of the input manifest.
-  - The `columns` field is set from the CSV header, if it is missing.
+- `--new-manifest-mode` *required*
```
Comment (jachym-tousek-keboola) [inline README.md]: Tbh not sure if this is the best name for the new flag... but not quite sure how to rename it... Maybe something like `--missing-schema-mode`.

Comment (michaljurecko) [reply README.md]: Yes, you are right, ... I suggest `--data-type-mode` (similar to the ENVs names bellow).

File: internal/pkg/processor/config/config.go [hunk 2]
```diff
@@ -51,6 +57,15 @@ func LoadConfig(configPath string) (cfg *Config, err error) {
 		return nil, kbc.UserErrorf("invalid configuration: %s", processJSONError(err))
 	}
 
+	switch dataTypeSupport {
+	case "hint":
+		conf.Parameters.NewManifestMode = slicerConfig.NewManifestModeColumns
+	case "authoritative":
+		conf.Parameters.NewManifestMode = slicerConfig.NewManifestModeSchema
+	default:
+		return nil, kbc.UserErrorf(`undexpected data type support: "%s"`, dataTypeSupport)
```
Comment (michaljurecko) [inline internal/pkg/processor/config/config.go:66]: This doesn't match specification, ENV can be undefined (or I did miss something):

image

Slack: https://keboolaglobal.slack.com/archives/C055HSMKX51/p1712570028466149?thread_ts=1712314989.537509&cid=C055HSMKX51

Comment (jachym-tousek-keboola) [reply internal/pkg/processor/config/config.go:66]: I think you missed the default value here: https://github.com/keboola/processor-split-table/blob/9641a467272d1aad64497cadf32f7a5547e1b787/internal/pkg/kbc/kbc.go#L51

Comment (michaljurecko) [reply internal/pkg/processor/config/config.go:66]: :+1:

File: README.md (hunk 1 above)
Comment (jachym-tousek-keboola) [reply README.md]: Done, see the new commits.

File: test/cli/dump-config-schema/args [hunk 3]
```diff
@@ -0,0 +1,8 @@
+--dump-config
```
Comment (michaljurecko) [inline test/cli/dump-config-schema/args]: I see no use-case for `dump-config-schema` test, because the field is not serialized to the JSON.

**Review by michaljurecko — APPROVED** [review]
LGTM

Please, use a better name for the flag, as you suggested.

File: test/cli/dump-config-schema/args (hunk 3 above)
Comment (jachym-tousek-keboola) [reply test/cli/dump-config-schema/args]: Hmm? I don't understand... the use-case is that the table has no input manifest and args is configured with `--data-type-mode schema`. This is supposed to create a manifest with the schema field, which it does.

Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: Yes, but it is covered by the `simple-no-schema` test, right?

Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: Original `dump-config` test was testing the `--dump` flag, so I don't know if we need two tests for the same thing.

Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: But that's a detail, if you want to leave it there, then ok.

Comment (jachym-tousek-keboola) [reply test/cli/dump-config-schema/args]: > Yes, but it is covered by the `simple-no-schema` test, right?

Well, no...
- `simple-no-schema` covers the case where manifest exists but doesn't have either `schema` or `columns`
- `dump-config-schema` covers the case where manifest doesn't exist at all

**Comment by michaljurecko** [comment]
:+1: Thanks for the explanation, so could we just give the test a better name?

`simple-no-schema` -> `mode-schema-empty-manifest`
`dump-config-schema` -> `mode-schema-missing-manifest`

File: test/cli/dump-config-schema/args (hunk 3 above)
Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: :+1: Thanks for the explanation, so could we just give the test a better name?

`simple-no-schema` -> `mode-schema-empty-manifest`
`dump-config-schema` -> `mode-schema-missing-manifest`

Or something similar.

Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: (maybe this creates a need to rename other tests too)

Comment (jachym-tousek-keboola) [reply test/cli/dump-config-schema/args]: I also renamed this:
`simple-empty-schema` -> `mode-schema-empty-manifest`
This test handles the case where manifest exists, has schema field, but the schema is empty array.

Comment (michaljurecko) [reply test/cli/dump-config-schema/args]: Thanks! :heavy_check_mark:

**Pull request merged**